metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java

```diff
@@ -46,35 +54,113 @@
         "provideMetadataForContributors",
         "provideMetadataForIdentifier",
         "provideMetadataForLanguage"
     })
     public void getCreatePublicationParsesHtmlAndReturnsMetadata(String testDescription, String html,
                                                                  CreatePublicationRequest expectedRequest)
         throws IOException {
-        String filename = "article.html";
-        URI uri = prepareWebServerAndReturnUriToMetadata(filename, html);
+        URI uri = prepareWebServerAndReturnUriToMetadata(ARTICLE_HTML, html);
         MetadataService metadataService = new MetadataService();
         Optional<CreatePublicationRequest> request = metadataService.getCreatePublicationRequest(uri);
 
-        assertThat(request.isPresent(), is(true));
-        CreatePublicationRequest actual = request.get();
+        CreatePublicationRequest actual = request.orElseThrow();
         actual.setContext(null);
 
         assertThat(actual, is(equalTo(expectedRequest)));
+    }
 
-        wireMockServer.stop();
+    @ParameterizedTest(name = "getCreatePublication returns date when date attribute {0} with value {1}")
+    @ArgumentsSource(DateArgumentsProvider.class)
+    void getCreatePublicationReturnsDateWhenDateVariantIsPresent(String tagAttribute, String date) throws IOException {
+        String html = generateHtmlWithSingleMetaTag(tagAttribute, date);
+        URI uri = prepareWebServerAndReturnUriToMetadata(ARTICLE_HTML, html);
+        MetadataService metadataService = new MetadataService();
+        Optional<CreatePublicationRequest> request = metadataService.getCreatePublicationRequest(uri);
+
+        CreatePublicationRequest actual = request.orElseThrow();
+        actual.setContext(null);
+
+        CreatePublicationRequest expectedRequest = getCreatePublicationRequestWithDateOnly(date);
+        assertThat(actual, is(equalTo(expectedRequest)));
+    }
+
+    @Test
+    @DisplayName("getCreatePublication accepts multiple dates, but only one publication date is returned")
+    void getCreatePublicationReturnsMostCompletePublicationDateWhenMultipleCandidatesArePresent() throws IOException {
+
+        List<MetaTagPair> metaDates = List.of(new MetaTagPair(DC_DATE, YEAR_ONLY),
+                new MetaTagPair(DC_DATE, FULL_DATE));
+
+        String html = createHtml(metaDates);
+        URI uri = prepareWebServerAndReturnUriToMetadata(ARTICLE_HTML, html);
+        MetadataService metadataService = new MetadataService();
+        Optional<CreatePublicationRequest> request = metadataService.getCreatePublicationRequest(uri);
+        CreatePublicationRequest actual = request.orElseThrow();
+        actual.setContext(null);
+
+        CreatePublicationRequest expectedRequest = getCreatePublicationRequestWithDateOnly(FULL_DATE);
+        assertThat(actual, is(equalTo(expectedRequest)));
+    }
+
+    @ParameterizedTest(name = "Dates that are shorter are accepted when bad date is {0}")
```

Comment:
The test name seems to contradict the method name

Reply:
Is the test checking that the year is picked up? i.e. the first number with 4 digits?

Reply:
I input a list containing a good date and a bad date, the test asserts that the good date is picked up. Do I need to make this clearer? (line 130, then line 140)

Reply:
The test name is a bit confusing. What impact does length have on good vs bad dates? Maybe test length logic in a different test?